tests/unit/dataactvalidator/test_b19_object_class_program_activity.py

```diff
@@ -0,0 +1,98 @@
+
+    op6 = ObjectClassProgramActivity(job_id=1, row_number=1, beginning_period_of_availa='1',
+                                     ending_period_of_availabil='1', agency_identifier='1',
+                                     allocation_transfer_agency='1', availability_type_code='2',
+                                     main_account_code='1', sub_account_code='1', object_class='1',
+                                     program_activity_code='1', by_direct_reimbursable_fun='r')
+
+    op7 = ObjectClassProgramActivity(job_id=1, row_number=1, beginning_period_of_availa='1',
+                                     ending_period_of_availabil='1', agency_identifier='1',
+                                     allocation_transfer_agency='1', availability_type_code='1',
+                                     main_account_code='2', sub_account_code='1', object_class='1',
+                                     program_activity_code='1', by_direct_reimbursable_fun='r')
+
+    op8 = ObjectClassProgramActivity(job_id=1, row_number=1, beginning_period_of_availa='1',
+                                     ending_period_of_availabil='1', agency_identifier='1',
+                                     allocation_transfer_agency='1', availability_type_code='1',
+                                     main_account_code='1', sub_account_code='2', object_class='1',
+                                     program_activity_code='1', by_direct_reimbursable_fun='r')
+
+    op9 = ObjectClassProgramActivity(job_id=1, row_number=1, beginning_period_of_availa='1',
+                                     ending_period_of_availabil='1', agency_identifier='1',
+                                     allocation_transfer_agency='1', availability_type_code='1',
+                                     main_account_code='1', sub_account_code='1', object_class='2',
+                                     program_activity_code='1', by_direct_reimbursable_fun='r')
+
+    op10 = ObjectClassProgramActivity(job_id=1, row_number=1, beginning_period_of_availa='1',
+                                     ending_period_of_availabil='1', agency_identifier='1',
+                                     allocation_transfer_agency='1', availability_type_code='1',
+                                     main_account_code='1', sub_account_code='1', object_class='1',
+                                     program_activity_code='2', by_direct_reimbursable_fun='r')
+
+    op11 = ObjectClassProgramActivity(job_id=1, row_number=1, beginning_period_of_availa='1',
+                                     ending_period_of_availabil='1', agency_identifier='1',
+                                     allocation_transfer_agency='1', availability_type_code='1',
+                                     main_account_code='1', sub_account_code='1', object_class='1',
+                                     program_activity_code='1', by_direct_reimbursable_fun='d')
+
```

Comment:
Since most of these fields are optional, it would probably be good to have some test cases where some of the values are missing.